src/PrivateDns/PrivateDns/Records/GetAzurePrivateDnsRecordSet.cs

```diff
@@ -0,0 +1,110 @@
+
+        [Parameter(Mandatory = true, ValueFromPipeline = true, HelpMessage = "The DnsZone object representing the zone in which to create the record set.", ParameterSetName = ObjectParameterSetName)]
+        [ValidateNotNullOrEmpty]
+        public PSPrivateDnsZone Zone { get; set; }
+
+        [Parameter(ParameterSetName = ResourceParameterSetName, Mandatory = true, ValueFromPipelineByPropertyName = true, HelpMessage = "Private DNS Zone ResourceID.")]
+        [ResourceIdCompleter("Microsoft.Network/privateDnsZones")]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(Mandatory = false, HelpMessage = "The name of the records in this record set (relative to the name of the zone and without a terminating dot).")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(Mandatory = false, ValueFromPipelineByPropertyName = true, HelpMessage = "The type of DNS records in this record set.")]
+        [ValidateNotNullOrEmpty]
+        public RecordType? RecordType { get; set; }
+
+        public override void ExecuteCmdlet()
+        {
+            string zoneName = null;
+            string resourceGroupName = null;
+
+            switch (this.ParameterSetName)
+            {
+                case FieldsParameterSetName:
+                    zoneName = this.ZoneName;
+                    resourceGroupName = this.ResourceGroupName;
+                    break;
+                case ResourceParameterSetName:
+                    PrivateDnsUtils.GetResourceGroupNameAndZoneNameFromResourceId(this.ParentResourceId, out resourceGroupName, out zoneName);
+                    break;
+                default:
+                    zoneName = this.Zone.Name;
+                    resourceGroupName = this.Zone.ResourceGroupName;
+                    break;
+            }
+
+            if (zoneName != null && zoneName.EndsWith("."))
+            {
+                zoneName = zoneName.TrimEnd('.');
+                this.WriteWarning($"Modifying zone name to remove terminating '.'.  Zone name used is \"{zoneName}\".");
+            }
+
+            if (this.Name != null)
+            {
+                if (this.RecordType == null)
+                {
+                    throw new PSArgumentException("If you specify the Name parameter you must also specify the RecordType parameter.");
+                }
+
+                var result = this.PrivateDnsClient.GetPrivateDnsRecordSet(this.Name, zoneName, resourceGroupName, this.RecordType.Value);
+                this.WriteObject(result);
+            }
+            else
+            {
+                List<PSPrivateDnsRecordSet> result = null;
+                result = this.RecordType == null ? this.PrivateDnsClient.ListRecordSets(zoneName, resourceGroupName) : this.PrivateDnsClient.ListRecordSets(zoneName, resourceGroupName, this.RecordType.Value);
+
+                foreach (var r in result)
```

Comment:
This should be
```
this.WriteObject(result, true)
```

This automatically takes care of enumerating lists and piping